### PR TITLE
feat: Add minimum OpenClaw version requirement and startup check (#307)

### LINKED ISSUE #307: Add minimum OpenClaw version requirement and startup check
### Summary
Add a minimum OpenClaw version requirement to hybrid-memory plugin, with a clear startup warning or error when running on an older version.

### Background
On 2026-03-10, Doris was running OpenClaw v2026.3.2 with hybrid-memory v2026.3.100. Several issues arose:
- The `hybrid-mem` CLI subcommand (registered by the plugin) didn't work — only available in OpenClaw ≥2026.3.8
- `openclaw plugins run ...` failed/hung on the older version
- Gateway SIGUSR1 hot-reload behavior differed between versions

### Proposed Changes

1. **Add `peerDependencies` to package.json:**
   ```json
   "peerDependencies": {
     "openclaw": ">=2026.3.8"
   }
   ```

2. **Add startup version check in the plugin init:**
   - Read the OpenClaw version from the gateway context/API
   - If below minimum, log a clear warning: `memory-hybrid: WARNING — OpenClaw v{current} detected, minimum recommended is v{min}. Some features (CLI subcommands, SIGUSR1 reload) may not work.`
   - Don't hard-fail — the plugin should still load for basic functionality

3. **Document version compatibility in README:**
   - v2026.3.100+ requires OpenClaw ≥2026.3.8
   - CLI subcommands require OpenClaw ≥2026.3.8
   - Basic plugin functionality works on ≥2026.3.0

### Version
v2026.3.100

### PR DESCRIPTION
**Pull request by markus-lassfolk**
## Summary

- Adds `peerDependencies: { "openclaw": ">=2026.3.8" }` to `package.json` to surface version incompatibility at install time
- New `utils/version-check.ts` with `parseVersion`, `isVersionAtLeast`, and `checkOpenClawVersion` utilities
- On plugin startup, `checkOpenClawVersion(api.version, api.logger)` emits a clear warning when the gateway is below v2026.3.8 — does **not** hard-fail so basic functionality still works
- Adds `version?: string` to the local `ClawdbotPluginApi` type stub (matches the real `OpenClawPluginApi` in the SDK)
- 17 new unit tests covering all version comparison edge cases

## Warning message format

```
memory-hybrid: WARNING — OpenClaw v2026.3.2 detected, minimum recommended is v2026.3.8. Some features (CLI subcommands, SIGUSR1 reload) may not work.
```

## Test plan

- [x] `npm test` — 118 test files, 2898 tests pass
- [x] `npx tsc --noEmit` — no TypeScript errors

Closes #307

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Low risk: adds a non-blocking startup version warning and packaging/docs updates; behavior change is limited to emitting warnings when the gateway is older or doesn’t expose `api.version`.
> 
> **Overview**
> Adds a minimum recommended OpenClaw gateway version (`v2026.3.8`) by declaring `openclaw` as a `peerDependencies` constraint and surfacing this requirement in the main README, Quickstart, CLI reference, and extension README.
> 
> Introduces `utils/version-check.ts` and wires it into plugin startup (`createPluginService.start`) to log a clear warning when `api.version` is missing or below the minimum (without hard-failing), updates the local SDK type stub to include `api.version`, and adds unit + integration tests to prevent regressions in the startup wiring and version parsing/comparison logic.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit fa172315aaf0fc2e2587ecc0a11361b67f83ac32. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: extensions/memory-hybrid/package.json [hunk 1]
```diff
@@ -34,6 +34,9 @@
     "lancedb",
     "embeddings"
   ],
+  "peerDependencies": {
```
Comment (markus-lassfolk) [inline extensions/memory-hybrid/package.json]: ### 🟡 Lockfile was not regenerated after adding the peer dependency

**Warning**

This PR changes the root package metadata, but `extensions/memory-hybrid/package-lock.json` still has the old top-level entry and does not record the new `peerDependencies`. That leaves the repo in a drifted state where `npm ci` / review tooling sees different metadata than `package.json`, and it also makes it easier for a later install to accidentally overwrite the intended compatibility signal. Please regenerate and commit the lockfile alongside this package manifest change.

File: extensions/memory-hybrid/utils/version-check.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * Logs a clear warning if the version is below the minimum — does not hard-fail.
+ *
+ * @param currentVersion - The `api.version` string passed by the gateway (may be undefined)
+ * @param logger - Plugin logger for emitting the warning
+ */
+export function checkOpenClawVersion(
+  currentVersion: string | undefined,
+  logger: { warn: (msg: string) => void },
+): void {
+  if (!currentVersion) return;
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/utils/version-check.ts]: Fixed. `checkOpenClawVersion` now emits a `logger.warn` when `currentVersion` is undefined, matching the suggested message. The test has been updated to expect this warning.

File: extensions/memory-hybrid/package.json [hunk 3]
```diff
@@ -34,6 +34,9 @@
...
+    "openclaw": ">=2026.3.8"
+  },
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/package.json]: Fixed. The `peerDependencies: { "openclaw": ">=2026.3.8" }` field has been added to the root package entry in `package-lock.json` so `npm ci` and lockfile tooling see consistent metadata.

File: extensions/memory-hybrid/package.json (hunk 1 above)
Comment (markus-lassfolk) [reply extensions/memory-hybrid/package.json]: Fixed. The `peerDependencies` field is now recorded in the root `""` entry of `package-lock.json`, resolving the drift between `package.json` and the lockfile.

File: extensions/memory-hybrid/setup/plugin-service.ts [hunk 4]
```diff
@@ -97,6 +98,8 @@ export function createPluginService(ctx: PluginServiceContext) {
         `memory-hybrid: initialized v${versionInfo.pluginVersion} (sqlite: ${sqlCount} facts, lance: ${resolvedLancePath}, model: ${cfg.embedding.model})`,
       );
 
+      checkOpenClawVersion(api.version, api.logger);
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/setup/plugin-service.ts]: Fixed. Added `extensions/memory-hybrid/tests/plugin-service-startup.test.ts` with three integration-style tests that call `createPluginService(ctx).start()` with real DB instances and assert on `api.logger.warn` calls: (1) undefined version emits warning, (2) below-minimum version emits warning, (3) at-minimum version emits no version warning.

File: extensions/memory-hybrid/tests/version-check.test.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    expect(isVersionAtLeast("not-a-version", "2026.3.8")).toBe(true);
+    expect(isVersionAtLeast("2026.3.8", "not-a-version")).toBe(true);
+  });
+});
+
+describe("checkOpenClawVersion", () => {
+  it("does nothing when version is undefined", () => {
+    const logger = { warn: vi.fn() };
+    checkOpenClawVersion(undefined, logger);
+    expect(logger.warn).not.toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/tests/version-check.test.ts]: Fixed. The test at L72-L77 now expects `logger.warn` to be called once and the message to contain `MIN_OPENCLAW_VERSION`, matching the correct behavior.

File: extensions/memory-hybrid/utils/version-check.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+export const MIN_OPENCLAW_VERSION = "2026.3.8";
+
+/**
+ * Parses a version string like "2026.3.8" into a numeric tuple.
+ * Returns null if the string cannot be parsed as three non-negative integers.
+ */
+export function parseVersion(version: string): [number, number, number] | null {
+  const parts = version.split(".");
+  if (parts.length !== 3) return null;
+  const nums = parts.map(Number);
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply extensions/memory-hybrid/utils/version-check.ts]: Fixed. The regex `^v?(\d+)\.(\d+)\.(\d+)` requires at least one digit between dots, so empty segments like `2026..8` no longer silently parse as `[2026, 0, 8]` — they return `null` as expected.
